float32 radius
float32 azimuthAngle
float32 elevationAngle
float32 speed
float32 intensity